### ISSUE TITLE
Unquote-splicing breaks map literals

### ISSUE THREAD

**Issue by NoahTheDuke** [clj-kondo/clj-kondo#2049]
**version**

2023.03.17

**macro usage**

N/A

**platform**

Ubuntu

**editor**

N/A

**problem**

Unquote-splicing doesn't work in map literals.

**repro**

```
$ echo "(def example {:a 1 #?@(:clj [:b 2 :c 3])})" | clj-kondo --lint -
<stdin>:1:20: error: missing value for key #?@ (:clj [:b 2 :c 3]) [:missing-map-value]
```

```
$ clj
Clojure 1.11.1
user=> (def example {:a 1 #?@(:clj [:b 2 :c 3])})
#'user/example
```

**expected behavior**

Minimal solution is that unquote splice should be recognized/ignored. Harder solution is to perform the unwrapping inline so the map literal is considered to have the branch-correct entries.

**Comment by borkdude** [comment]
```
$ echo "(def example {:a 1 #?@(:clj [:b 2 :c 3])})" | clj-kondo --lint - --lang cljc
linting took 15ms, errors: 0, warnings: 0
```

You need to set the language when linting from stdin.

**Comment by NoahTheDuke** [comment]
Oops, thanks.

**Comment by borkdude** [comment]
@NoahTheDuke To be fair, I think clj-kondo should warn on using a conditional in a non-cljc file!
